generators/server/templates/src/main/java/package/service/UserService.java.ejs

```diff
@@ -1042,8 +1044,6 @@ public class UserService {
         }
         if (details.get("email") != null) {
             user.setEmail(((String) details.get("email")).toLowerCase());
-        } else {
```

Comment:
It was added in https://github.com/jhipster/generator-jhipster/pull/9947 Can you confirm that flow is not affected?

Reply:
In my opinion mapping the `sub` claim to the email is wrong. This [claim is an Identifier](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims).

In ADFS it looks similar to a base64 encoded string. Since the `email` claim is not provided by default in ADFS this leads to the validation error (see description).

However I did not try that with Okta, so I can't confirm.

Reply:
Unless you are willing to verify the said flow, I shall prefer to keep it as is.